shapely/strtree.py

```diff
@@ -122,28 +129,59 @@ def __reduce__(self):
         else:
             return (STRtree, (self.geometries,))
 
-    def query_items(self, geom: BaseGeometry) -> Sequence[Any]:
-        """Query for nodes which intersect the geom's envelope to get
-        stored items.
+    def query_items(
+        self, geom: BaseGeometry, predicate=None, distance=None
+    ) -> Sequence[Any]:
+        """
+        Return the index of all geometries in the tree with extents that
+        intersect the envelope of the input geometry.
```

Comment:
Strictly speaking, the original sentence was more correct. Shapely's r-tree is more general that pygeos was. We can store values that are not indexes of anything.

```suggestion
        Return stored items from tree nodes that intersect the envelope of the input geometry, optionally filtering them by a spatial predicate involving the input geometry.
```

Reply:
I will try to update the docstrings a bit based on your feedback, but I would prefer to keep some simplified version. For example, I am not sure that the "from tree nodes" adds value for this first sentence (that seems an implementation detail?), and rather makes it harder to understand. What matters to the user is that it returns the items that correspond to the geometries that intersect with the envelope, I think?

Reply:
Not sure if this helps, but tries to help better tease apart the distinction between index vs item / arbitrary value here:

`Return items from the tree that intersect with the envelope of the input geometry.  Items may be identified by an integer index (default) or arbitrary item values (optional) if those are provided when constructing the tree.  Results can be optionally filtered by a spatial predicate...`